### PR TITLE
Fix for Remove prompt pulse bug

### PR DESCRIPTION
**Pull request by dmitry-ganyushin**
### Description of work

This PR should fix issues for [RemovePromptPulse](https://docs.mantidproject.org/nightly/algorithms/RemovePromptPulse-v1.html) to mask correctly the first pulse including the case of ToFmin <0. Changes were manually merged. If a data file has a minimum time-of-flight which is negative, this caused issues for the [RemovePromptPulse](https://docs.mantidproject.org/nightly/algorithms/RemovePromptPulse-v1.html) which ignored the first pulse completely, and that was fixed.

Duplicate for https://github.com/mantidproject/mantid/pull/37932

### CODE REVIEW

**Comment by sf1919** [comment]
Should #37932 be closed? I'm not sure why there is this duplicate. The other one is currently pointed at `release-next` to go into v6.11 but this one is pointed at `main` instead. As `release-next` gets merged into main it doesn't make sense to have separate PRs for `release-next` and `main`

**Review by peterfpeterson — CHANGES_REQUESTED** [review]
This needs release notes in the v6.12 section. You'll need to cherry-pick https://github.com/mantidproject/mantid/pull/37968/commits/43e26eb6289fe03f9f9936a7a9a71d1df614e5f9 first

**Comment by dmitry-ganyushin** [comment]
> This needs release notes in the v6.12 section. You'll need to cherry-pick [43e26eb](https://github.com/mantidproject/mantid/commit/43e26eb6289fe03f9f9936a7a9a71d1df614e5f9) first

This commit is to understand the failure of 2 system tests. It will be decorated later.

File: Framework/WorkflowAlgorithms/test/AlignAndFocusPowderTest.h [hunk 1]
```diff
@@ -786,9 +786,9 @@ class AlignAndFocusPowderTest : public CxxTest::TestSuite {
     if (m_compressStartTime != "0")
       align_and_focus.setProperty("CompressStartTime", m_compressStartTime);
 
-    // Remove prompt pulse; will cutoff last 6 long-TOF peaks (freq is 200 Hz)
+    // Remove prompt pulse; will cutoff the first peak
     if (m_removePromptPulse)
-      align_and_focus.setProperty("RemovePromptPulseWidth", 1e4);
+      align_and_focus.setProperty("RemovePromptPulseWidth", 2200.0);
```
Comment (peterfpeterson) [inline Framework/WorkflowAlgorithms/test/AlignAndFocusPowderTest.h:791]: Why are you changing the input parameter?

File: Framework/WorkflowAlgorithms/test/AlignAndFocusPowderTest.h [hunk 2]
```diff
@@ -786,9 +786,9 @@ class AlignAndFocusPowderTest : public CxxTest::TestSuite {
     if (m_compressStartTime != "0")
       align_and_focus.setProperty("CompressStartTime", m_compressStartTime);
 
-    // Remove prompt pulse; will cutoff last 6 long-TOF peaks (freq is 200 Hz)
+    // Remove prompt pulse; will cutoff the first peak
```
Comment (peterfpeterson) [inline Framework/WorkflowAlgorithms/test/AlignAndFocusPowderTest.h:789]: Please combine these two comments because both are useful

**Review by peterfpeterson — CHANGES_REQUESTED** [review]
Please put more information into the description so people know what is going on without going to a second closed PR. Leave the link to the original in case people do want to go digging.

File: Framework/WorkflowAlgorithms/test/AlignAndFocusPowderTest.h (hunk 1 above)
Comment (dmitry-ganyushin) [reply Framework/WorkflowAlgorithms/test/AlignAndFocusPowderTest.h:791]: Previously, it worked in a way that in fact all pulses were not filtered. Now  the first pulse which is ended on ~2200 is filtered out. I printed out the date and ensured that the first pulse is filtered out.

File: Framework/WorkflowAlgorithms/test/AlignAndFocusPowderTest.h (hunk 2 above)
Comment (dmitry-ganyushin) [reply Framework/WorkflowAlgorithms/test/AlignAndFocusPowderTest.h]: Modified as requested

File: Framework/WorkflowAlgorithms/test/AlignAndFocusPowderTest.h (hunk 1 above)
Comment (peterfpeterson) [reply Framework/WorkflowAlgorithms/test/AlignAndFocusPowderTest.h:791]: What is the minimum TOF in the events for this data? My concern is that you greatly reduced the window from 0->10000us to 0->2200us. What is in the range between? An image would be really helpful.

Comment (dmitry-ganyushin) [reply Framework/WorkflowAlgorithms/test/AlignAndFocusPowderTest.h:791]: RemovePromptPulseBug
RemovePromptPulse
Here is a version with the bug, and fixed. Period is 5000 microseconds, and the width is 2200 just to  cut out the first pulse.

**Comment by dmitry-ganyushin** [comment]
Here is a version with the bug, and fixed. Period is 5000 microseconds, and the width is 2200 just to cut out the first pulse
RemovePromptPulseBug
RemovePromptPulse
.

**Review by peterfpeterson — APPROVED** [review]

**Pull request merged**